source/agent/conference/conference.js

```diff
@@ -1025,21 +1063,24 @@ var Conference = function (rpcClient, selfRpcId) {
     } else {
       var origin = participants[participantId].getOrigin();
       var format_preference;
-      if (pubInfo.type === 'webrtc') {
-        const rtcPubInfo = translateRtcPubIfNeeded(pubInfo);
-        // Set formatPreference
-        rtcPubInfo.tracks.forEach(track => {
-          track.formatPreference = {optional: room_config.mediaIn[track.type]};
-        });
-        initiateStream(streamId, {owner: participantId, type: pubInfo.type, origin});
-        return rtcController.initiate(participantId, streamId, 'in', participants[participantId].getOrigin(), rtcPubInfo)
-        .then((result) => {
-          callback('callback', result);
-        })
-        .catch((e) => {
-          removeStream(streamId);
-          callback('callback', 'error', e.message ? e.message : e);
-        });
+      if (pubInfo.type === 'webrtc' || pubInfo.type === 'quic') {
+          const controller = pubInfo.type === 'webrtc' ? rtcController : quicController;
+          const rtcPubInfo = translateRtcPubIfNeeded(pubInfo);
+          if (rtcPubInfo.tracks) {
+              // Set formatPreference
+              rtcPubInfo.tracks.forEach(track => {
```

Comment:
I don't see a reason you place the track format preference when pubInfo type is quic.